relayer/pkg/chainlink/relay.go

```diff
@@ -3,34 +3,31 @@ package chainlink
 import (
 	"context"
 	"encoding/json"
+	"errors"
+	"fmt"
 
-	"github.com/pkg/errors"
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+	relaytypes "github.com/smartcontractkit/chainlink-common/pkg/types"
 
 	starkchain "github.com/smartcontractkit/chainlink-starknet/relayer/pkg/chainlink/chain"
 	"github.com/smartcontractkit/chainlink-starknet/relayer/pkg/chainlink/ocr2"
-
-	"github.com/smartcontractkit/chainlink-common/pkg/logger"
-	relaytypes "github.com/smartcontractkit/chainlink-common/pkg/types"
 )
 
 var _ relaytypes.Relayer = (*relayer)(nil) //nolint:staticcheck
 
 type relayer struct {
-	chain starkchain.Chain
-	ctx   context.Context
+	chain  starkchain.Chain
+	stopCh services.StopChan
```

Comment:
Is there a reason you prefer a channel over ctx? With ctx we can easily stack ctxs on top of each other.

also, I just noticed that ctx (and stopCh) aren't being used 🤣 . Do other chain relayer pass them into the providers?  I notice solana[ attempts to](https://github.com/smartcontractkit/chainlink-solana/blob/develop/pkg/solana/relay.go), at least passes it to the configprovider constructor...but it remains unused.

Reply:
i see ~ currently reading https://go.dev/blog/context-and-structs 🔍